OrbitGl/GraphTrack.cpp

```diff
@@ -37,30 +36,17 @@ void GraphTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
 
   Box box(pos_, Vec2(size_[0], -size_[1]), track_z);
 
-  // Draw label with graph value at current mouse position.
-  const Color kWhiteColor(255, 255, 255, 255);
-  const Color kBlackColor(0, 0, 0, 255);
-  double graph_value = GetValueAtTime(time_graph_->GetCurrentMouseTimeNs());
-
-  int white_text_box_width =
-      canvas->GetTextRenderer().GetStringWidth(std::to_string(graph_value).c_str());
-
-  Vec2 white_text_box_size(white_text_box_width, layout.GetTextBoxHeight());
-  auto rightmost_x_position = canvas->GetWorldTopLeftX() + canvas->GetWorldWidth() -
-                              layout.GetRightMargin() - layout.GetSliderWidth() -
-                              white_text_box_size[0];
-  Vec2 white_text_box_position(
-      std::min(canvas->GetMouseX(), rightmost_x_position),
-      y0 - static_cast<float>((max_ - graph_value) * size_[1] / value_range_) -
-          white_text_box_size[1] / 2.f);
-
   if (!picking) {
-    canvas->GetTextRenderer().AddText(
-        std::to_string(graph_value).c_str(), white_text_box_position[0],
-        white_text_box_position[1] + layout.GetTextOffset(), GlCanvas::kZValueTextUi, kBlackColor,
-        time_graph_->CalculateZoomedFontSize(), white_text_box_size[0]);
-    Box white_text_box(white_text_box_position, white_text_box_size, GlCanvas::kZValueUi);
-    batcher->AddBox(white_text_box, kWhiteColor);
+    double graph_value;
+    uint64_t graph_time;
+    bool is_there_a_value =
+        GetPreviousValue(time_graph_->GetCurrentMouseTimeNs(), &graph_time, &graph_value);
```

Comment:
"is_there_a_value" is not a great name, "has_value_at_time" maybe? Actually, I think the interface would be cleaner here is we used a std::optional. The function could return a std::optional<std::pair<uint64_t, double>> for example. We could then replace "GetPreviousValue" by "GetPointForTime(...)" maybe, wdyt?

Reply:
GetLastPointBeforeTime

Reply:
GetPointForTime maybe is a little misleading, because is returning a time (related with x-coordinate) but a value who doesn't seem to have relation with y-position. I called it GetLastMeasure, what do you think?